File: src/main/java/com/ptmanager/backend/config/DataSeeder.java
package com.ptmanager.backend.config;

import java.time.LocalDate;
import java.time.LocalTime;

import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import com.ptmanager.backend.domain.Shift;
import com.ptmanager.backend.domain.User;
import com.ptmanager.backend.domain.UserRole;
import com.ptmanager.backend.domain.Workplace;
import com.ptmanager.backend.repository.ShiftRepository;
import com.ptmanager.backend.repository.UserRepository;
import com.ptmanager.backend.repository.WorkplaceRepository;

@Component
public class DataSeeder implements CommandLineRunner {

    private final WorkplaceRepository workplaceRepository;
    private final UserRepository userRepository;
    private final ShiftRepository shiftRepository;

    public DataSeeder(WorkplaceRepository workplaceRepository,
                      UserRepository userRepository,
                      ShiftRepository shiftRepository) {
        this.workplaceRepository = workplaceRepository;
        this.userRepository = userRepository;
        this.shiftRepository = shiftRepository;
    }

    @Override
    public void run(String... args) {
        if (userRepository.count() > 0) {
            return;
        }

        Workplace workplace = workplaceRepository.save(
            new Workplace(null, "PT Manager Cafe", "Seoul Gangnam-gu", "CAFE01"));

        User employee = userRepository.save(
            new User(null, "employee@ptmanager.test", "Kim Employee",
                UserRole.EMPLOYEE, workplace.getId(), 12000));
        userRepository.save(
            new User(null, "employer@ptmanager.test", "Park Employer",
                UserRole.EMPLOYER, workplace.getId(), 0));

        LocalDate today = LocalDate.now();
        shiftRepository.save(new Shift(null, workplace.getId(), employee.getId(),
            today, LocalTime.of(9, 0), LocalTime.of(14, 0), false));
        shiftRepository.save(new Shift(null, workplace.getId(), employee.getId(),
            today.plusDays(1), LocalTime.of(14, 0), LocalTime.of(20, 0), false));
    }
}

File: src/main/kotlin/com/ptmanager/backend/config/DataSeeder.kt
package com.ptmanager.backend.config

import com.ptmanager.backend.domain.Shift
import com.ptmanager.backend.domain.User
import com.ptmanager.backend.domain.UserRole
import com.ptmanager.backend.domain.Workplace
import com.ptmanager.backend.repository.ShiftRepository
import com.ptmanager.backend.repository.UserRepository
import com.ptmanager.backend.repository.WorkplaceRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import java.time.LocalDate
import java.time.LocalTime

@Component
class DataSeeder(
    private val workplaceRepository: WorkplaceRepository,
    private val userRepository: UserRepository,
    private val shiftRepository: ShiftRepository,
) : CommandLineRunner {

    override fun run(vararg args: String?) {
        if (userRepository.count() > 0) {
            return
        }

        val workplace = workplaceRepository.save(
            Workplace(name = "PT Manager Cafe", address = "Seoul Gangnam-gu", inviteCode = "CAFE01"),
        )

        val employee = userRepository.save(
            User(
                email = "employee@ptmanager.test",
                password = "password",
                name = "Kim Employee",
                role = UserRole.EMPLOYEE,
                workplaceId = workplace.id,
                hourlyWage = 12000,
            ),
        )
        userRepository.save(
            User(
                email = "employer@ptmanager.test",
                password = "password",
                name = "Park Employer",
                role = UserRole.EMPLOYER,
                workplaceId = workplace.id,
                hourlyWage = 0,
            ),
        )

        val today = LocalDate.now()
        shiftRepository.save(
            Shift(
                workplaceId = workplace.id!!,
                employeeId = employee.id!!,
                workDate = today,
                startTime = LocalTime.of(9, 0),
                endTime = LocalTime.of(14, 0),
            ),
        )
        shiftRepository.save(
            Shift(
                workplaceId = workplace.id!!,
                employeeId = employee.id!!,
                workDate = today.plusDays(1),
                startTime = LocalTime.of(14, 0),
                endTime = LocalTime.of(20, 0),
            ),
        )
    }
}
